test/test_um2netcdf.py

```diff
@@ -744,6 +753,57 @@ def test_fix_level_coord_skipped_if_no_levels(z_sea_rho_data, z_sea_theta_data):
     um2nc.fix_level_coord(m_cube, z_sea_rho_data, z_sea_theta_data)
 
 
+# tests - fix pressure level data
+
+def test_fix_pressure_levels_no_pressure_coord(get_fake_cube_coords):
+    cube = get_fake_cube_coords()
+
+    with pytest.raises(iris.exceptions.CoordinateNotFoundError):
+        cube.coord("pressure")  # ensure missing 'pressure' coord
+
+    um2nc.fix_pressure_levels(cube)  # should just exit
+
+
+def _add_attrs_points(m_plevs: mock.MagicMock, points):
+    # NB: iris attributes appear to be added via mixins, so it's easier but
+    #     less desirable to rely on mock attrs here
+    setattr(m_plevs, "attributes", {"positive": None})
+    setattr(m_plevs, "points", points)
+
+
+def test_fix_pressure_levels_do_rounding(get_fake_cube_coords):
+    m_pressure = mock.Mock()
+    _add_attrs_points(m_pressure, [1.000001, 0.000001])
+    extra = {"pressure": m_pressure}
+    cube = get_fake_cube_coords(extra)
+
+    um2nc.fix_pressure_levels(cube)
+
+    # TODO: test flaw, this verifies pressure coord but ignores fix_pressure_levels()
+    #       returning a new cube if the pressure is reversed. This is verified
+    #       in command line testing though
+    c_pressure = cube.coord('pressure')
+    assert c_pressure.attributes["positive"] == "down"
+    assert all(c_pressure.points == [1.0, 0.0])
+
+
+def test_fix_pressure_levels_reverse_pressure(get_fake_cube_coords):
+    m_pressure = mock.Mock()
+    _add_attrs_points(m_pressure, [0.000001, 1.000001])
+    extra = {"pressure": m_pressure}
+    cube = get_fake_cube_coords(extra)
+
+    with mock.patch("iris.util.reverse"):
+        um2nc.fix_pressure_levels(cube)
+
+    # TODO: test flaw, this verifies pressure coord but ignores fix_pressure_levels()
+    #       returning a new cube if the pressure is reversed. This is verified
+    #       in command line testing though
+    c_pressure = cube.coord('pressure')
```

Comment:
Is it worth checking whether the mocked `iris.util.reverse()` is called here, or does that not really add anything?

Reply:
Sounds like a good idea!